integration-test.sh

```diff
@@ -543,12 +543,14 @@ function main {
 
 check_prerequisites
 
-# allow invocation from release-docker.sh
-if [[ $1 == "test-docker-release" ]]; then
-    test_docker_release_image
+trap "tear_down" EXIT
+
+# if argument is the name of a function, set up and call it
+if declare -f "$1" > /dev/null
+then
+    set_up
+    $1
```

Comment:
Since there are various helper functions in this script and maybe it's time consuming to scan for what is a real test, does it make sense that we store a whitelist with the functions we are allowed to call, currently: `test_configure,test_list,test_distribution_fails_with_wrong_track_params,test_download,test_sources,test_distributions,test_benchmark_only,test_docker_dev_image,test_docker_release_image`? We could also rely on the convention, like Python's testunit, that all callable functions start with `test_`.

If we do this, we could simplify the code in [run_test()](https://github.com/elastic/rally/blob/9b9c7e03b79a11414740fa8f543b7a58b162149b/integration-test.sh#L493-L512) to just loop through the whitelist.

Reply:
Personally, it doesn't feel to me like the protection we get (prevent users from calling utility functions individually?) is worth the additional maintenance burden, so I will merge as is.